internal/pkg/cli/app_init.go

```diff
@@ -240,8 +239,10 @@ func (o *initAppOpts) isDomainOwned() error {
 	}
 	var errDomainNotFound *route53.ErrDomainNotFound
 	if errors.As(err, &errDomainNotFound) {
-		o.isDomainNotFoundInAccount = true
-		return nil
+		log.Errorln("The account does not seem to own the domain that you entered. Please make sure that the " +
```

Comment:
nit: can we ensure that the text stays within 80 characters as much as possible so that it renders nicely for the default terminal screen size? https://github.com/aws/copilot-cli/blob/mainline/STYLE_GUIDE.md#screen-sizes 
```go
log.Errorf(`The account does not seem to own the domain that you entered.
Please make sure that %s is registered with Route53 in your account.
To transfer domain registration to Route53, see:
https://docs.aws.amazon.com/Route53/latest/DeveloperGuide/domain-transfer-to-route-53.html
`, o.domainName)

```